prow/gerrit/client/syncer.go

```diff
@@ -37,40 +37,46 @@ type opener interface {
 }
 
 type SyncTime struct {
-	val    LastSyncState
-	lock   sync.RWMutex
-	path   string
-	opener opener
-	ctx    context.Context
+	val            LastSyncState
+	lock           sync.RWMutex
+	path           string
+	opener         opener
+	ctx            context.Context
+	stateChan      chan LastSyncState
+	forceFlushChan chan bool
+	uploadInterval time.Duration
```

Comment:
Nit: I get a sense that these fields should live in a separate struct. If you considered this and rejected it, I'd like to know why (add in commit message description?).

Reply:
Tried to refactor this into something like:
```
type SyncTimeController struct {
  <SOME OTHER FIELDS>
  uploadInterval time.Duration
  syncTime *SyncTime
}
```

Then realized that even `SyncTime` is a wrapper around `LastSyncState`, and it doesn't add that much value to have one more layer